### PR TITLE
Qrcode scanning and display

### PR DESCRIPTION
**Pull request by or-else**
Does not work in the simulator. Tested on a real device, works OK.

### CODE REVIEW

File: TinodeSDK.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -540,10 +540,10 @@
 		0A431BE62244A0BE00A837F7 /* Release */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
-				CODE_SIGN_IDENTITY = "";
+				CODE_SIGN_IDENTITY = "Apple Development";
```
Comment (aforge) [inline TinodeSDK.xcodeproj/project.pbxproj:543]: Remove these?

File: Tinodios/AddByIDViewController.swift [hunk 2]
```diff
@@ -45,13 +69,49 @@ class AddByIDViewController: UIViewController {
...
+        showCodeButton.backgroundColor = UIColor.link
+        scanCodeButton.tintColor = UIColor.link
+        scanCodeButton.backgroundColor = UIColor.systemBackground
+    }
+
+    @IBAction func scanCodePressed(_ sender: Any) {
+        cameraPreviewView.isHidden = false
+        qrcodeImageView.isHidden = true
+
+        scanQRCode()
```
Comment (aforge) [inline Tinodios/AddByIDViewController.swift:104]: Please show a toast (error message) if this function call fails.

File: Tinodios/AddByIDViewController.swift [hunk 3]
```diff
@@ -77,4 +137,89 @@ class AddByIDViewController: UIViewController {
...
+
+    func metadataOutput(_ output: AVCaptureMetadataOutput, didOutput metadataObjects: [AVMetadataObject], from connection: AVCaptureConnection) {
+        captureSession.stopRunning()
+
+        if let metadataObject = metadataObjects.first {
+            guard let readableObject = metadataObject as? AVMetadataMachineReadableCodeObject else { return }
+            guard let stringValue = readableObject.stringValue else { return }
+            AudioServicesPlaySystemSound(SystemSoundID(kSystemSoundID_Vibrate))
+            if stringValue.hasPrefix(AddByIDViewController.kTopicUriPrefix) {
+                let swiftSucks = stringValue.index(stringValue.startIndex, offsetBy: AddByIDViewController.kTopicUriPrefix.count)
```
Comment (aforge) [inline Tinodios/AddByIDViewController.swift:219]: Use `String.dropFirst()` to make things easier.

File: TinodiosDB.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -359,8 +359,10 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = AEE0CC9C2915B09EB3E7D8E1 /* Pods-TinodiosDB.debug.xcconfig */;
 			buildSettings = {
+				CODE_SIGN_IDENTITY = "Apple Development";
```
Comment (aforge) [inline TinodiosDB.xcodeproj/project.pbxproj:362]: Remove this.

**Review by aforge — APPROVED** [review]

File: Tinodios/AddByIDViewController.swift (hunk 3 above)
Comment (or-else) [reply Tinodios/AddByIDViewController.swift]: Done

File: TinodeSDK.xcodeproj/project.pbxproj (hunk 1 above)
Comment (or-else) [reply TinodeSDK.xcodeproj/project.pbxproj:543]: I don't see how. It's added by XCode automatically.

File: TinodiosDB.xcodeproj/project.pbxproj (hunk 4 above)
Comment (or-else) [reply TinodiosDB.xcodeproj/project.pbxproj:362]: Like before, I don't see how short of editing it by hand.

File: Tinodios/AddByIDViewController.swift (hunk 2 above)
Comment (or-else) [reply Tinodios/AddByIDViewController.swift:104]: done

**Pull request merged**